src/Connection.php

```diff
@@ -196,7 +197,45 @@ protected static function connectDbalConnection(array $dsn)
 
         $connectionParams = ['pdo' => $pdo];
 
-        return DriverManager::getConnection($connectionParams);
+        $connection = DriverManager::getConnection($connectionParams);
+
+        // Oracle CLOB/BLOB has limited SQL support, see:
+        // https://stackoverflow.com/questions/12980038/ora-00932-inconsistent-datatypes-expected-got-clob#12980560
+        // fix this Oracle inconsistency by using VARCHAR/VARBINARY instead (but limited to 4000 bytes)
+        if ($connection->getDatabasePlatform() instanceof OraclePlatform) {
+            \Closure::bind(function () use ($connection) {
+                $connection->platform = new class() extends OraclePlatform {
+                    private function forwardTypeDeclarationSQL(string $targetMethodName, array $column): string
+                    {
+                        $backtrace = debug_backtrace(DEBUG_BACKTRACE_PROVIDE_OBJECT | DEBUG_BACKTRACE_IGNORE_ARGS);
+                        foreach ($backtrace as $frame) {
+                            if ($this === ($frame['object'] ?? null)
+                                && $targetMethodName === ($frame['function'] ?? null)) {
+                                throw new Exception('Long CLOB/TEXT (4000+ bytes) is not supported for Oracle');
```

Comment:
I still don't see the reason why it's not supported. CLOB definitely can be bigger than 4000 chars.
And what about BLOB?

Reply:
Because https://stackoverflow.com/questions/12980038/ora-00932-inconsistent-datatypes-expected-got-clob#12980560, Oracle simply does not allow CLOB/BLOB types to be used in SQL directly, even if bound inside queiries like `UPDATE t SET x = []`.